pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -738,3 +741,42 @@ func (u *upgradeNgAlerting) updateAlertmanagerFiles(orgId int64, migrator *migra
 func (u *upgradeNgAlerting) SQL(migrator.Dialect) string {
 	return "code migration"
 }
+
+// CheckUnifiedAlertingEnabledByDefault determines the final status of unified alerting, if it is not enabled explicitly.
```

Comment:
What happens here if a customer has been migrated to Unified Alerting but also has alerts in the `alerts` table (as the upgrade does not delete the `alerts` table?)

Reply:
If the legacy table has alerts then the only way for a user to switch to ngalert is to explicitly opt-in. If they do, then this will not run at all.